### ISSUE TITLE
Support `extends` and `plugins`

### ISSUE THREAD

**Issue by kentcdodds** [kentcdodds/eslint-find-new-rules#1]
If a config `extends` another, I'd like it to resolve that to determine the rules that are actually covered by the config.

**Comment by ta2edchimp** [comment]
I had a quick look into this and also [found a working solution](https://github.com/ta2edchimp/eslint-find-new-rules/compare/master...ta2edchimp:support-extends).

Problem now is, that `npm run validate` warns that the set threshold for uncovered loc has been underrun with these changes; when I add another test (using the new `getRules` function for example) or just a very basic, simple test like the following:

``` JavaScript
test('provokes a memory leak warning', t => {
  const rules = [1, 2]
  t.same(rules, [1, 2, 3])
})
```

I receive the following warning and the counter of passed tests goes crazy:

```
(node) warning: possible EventEmitter memory leak detected. 11 test listeners added. Use emitter.setMaxListeners() to increase limit.
```

Atm I assume an issue with ava :S (for the record, I'm using node `5.8.0`).

**Comment by kentcdodds** [comment]
I actually saw that error earlier when trying to pull the `bin` test into a different file. I just figured it was something weird going on with what `bin` was doing with the dynamic require...

Why don't you submit that as a PR and we can continue discussion there :+1:

**Comment by sarbbottam** [comment]
I have been wondering the same, how `extend` is being resolved at https://github.com/kentcdodds/eslint-find-new-rules/blob/master/bin.js#L24-L26

**Comment by kentcdodds** [comment]
Yeah, right now it's not being resolved at all. But if you'd like to participate in the discussion, see: https://github.com/eslint/eslint/issues/5600

**Comment by sarbbottam** [comment]
Sounds good! Thanks for the heads up.

**Comment by kentcdodds** [comment]
The `extends` part was merged. Now we need to support `plugins`. I probably shouldn't have closed #14 because that actually turns out to be more work. I'm going to re-open that one to do discussion it that :+1:
